Lib/test/test_doctest/test_doctest.py

```diff
@@ -2944,235 +2940,236 @@ def test_unicode(): """
-not stderr:
-
-    >>> rc1, err1
-    (1, b'')
-    >>> print(normalize(out1))                # doctest: +ELLIPSIS
-    **********************************************************************
-    File "...myfile.doc", line 4, in myfile.doc
-    Failed example:
-        "abcdef"
-    Expected:
-        'a...f'
-    Got:
-        'abcdef'
-    **********************************************************************
-    File "...myfile.doc", line 6, in myfile.doc
-    Failed example:
-        "ajkml"
-    Expected:
-        'a...l'
-    Got:
-        'ajkml'
-    **********************************************************************
-    1 items had failures:
-       2 of   3 in myfile.doc
-    ***Test Failed*** 2 failures.
-
-With -o ELLIPSIS specified, the second run, against just the first file, should
-produce no errors, and with -o NORMALIZE_WHITESPACE also specified, neither
-should the third, which ran against both files:
-
-    >>> rc2, out2, err2
-    (0, b'', b'')
-    >>> rc3, out3, err3
-    (0, b'', b'')
-
-The fourth run uses FAIL_FAST, so we should see only one error:
-
-    >>> rc4, err4
-    (1, b'')
-    >>> print(normalize(out4))                # doctest: +ELLIPSIS
-    **********************************************************************
-    File "...myfile.doc", line 4, in myfile.doc
-    Failed example:
-        "abcdef"
-    Expected:
-        'a...f'
-    Got:
-        'abcdef'
-    **********************************************************************
-    1 items had failures:
-       1 of   2 in myfile.doc
-    ***Test Failed*** 1 failures.
-
-The fifth test uses verbose with the two options, so we should get verbose
-success output for the tests in both files:
-
-    >>> rc5, err5
-    (0, b'')
-    >>> print(normalize(out5))
-    Trying:
-        1 + 1
-    Expecting:
-        2
-    ok
-    Trying:
-        "abcdef"
-    Expecting:
-        'a...f'
-    ok
-    Trying:
-        "ajkml"
-    Expecting:
-        'a...l'
-    ok
-    1 items passed all tests:
-       3 tests in myfile.doc
-    3 tests in 1 items.
-    3 passed and 0 failed.
-    Test passed.
-    Trying:
-        1 + 1
-    Expecting:
-        2
-    ok
-    Trying:
-        "abc   def"
-    Expecting:
-        'abc def'
-    ok
-    1 items had no tests:
-        myfile2
-    1 items passed all tests:
-       2 tests in myfile2.test_func
-    2 tests in 2 items.
-    2 passed and 0 failed.
-    Test passed.
-
-We should also check some typical error cases.
-
-Invalid file name:
-
-    >>> rc, out, err = script_helper.assert_python_failure(
-    ...         '-m', 'doctest', 'nosuchfile')
-    >>> rc, out
-    (1, b'')
-    >>> # The exact error message changes depending on the platform.
-    >>> print(normalize(err))                    # doctest: +ELLIPSIS
-    Traceback (most recent call last):
-      ...
-    FileNotFoundError: [Errno ...] ...nosuchfile...
+if support.has_subprocess_support:
```

Comment:
What I don't like is that the diff is huge. Maybe we can do something like

```python
def doctest_requires_subproccess(func):
    if not support.has_subprocess_support:
        func.__doc__ = None  # skip the doctest by setting it to `None`
    return func

@doctest_requires_subprocess
def test_CLI():  # as before, no changes
    ...
```

I've tried this locally with `if True`:

```
» ./python.exe -m test test_doctest -m test_CLI
Using random seed: 4265888589
0:00:00 load avg: 2.23 Run 1 test sequentially
0:00:00 load avg: 2.23 [1/1] test_doctest
test_doctest ran no tests

== Tests result: NO TESTS RAN ==

1 test run no tests:
    test_doctest

Total duration: 59 ms
Total tests: run=0 (filtered)
Total test files: run=1/1 (filtered) run_no_tests=1
Result: NO TESTS RAN
```

Reply:
Good idea; I've generalized it slightly so it can accept any skip condition.